feature-libs/product-multi-dimensional/selector/core/services/product-multi-dimensional-selector.service.ts

```diff
@@ -157,7 +157,7 @@ export class ProductMultiDimensionalSelectorService {
   protected createVariantCategoryGroup(
     variantMatrix: VariantMatrixElement[]
   ): VariantCategoryGroup {
-    const parentVariantCategory = variantMatrix[0].parentVariantCategory;
+    const parentVariantCategory = variantMatrix[0]?.parentVariantCategory;
```

Comment:
`query:` Are we sure at this stage that `variantMatrix` is not null and there is an object at index 0?

Reply:
good catch, I thought, I've fixed that.

Reply:
I've changed the name to elements as variantMatrix refers to the whole tree
```typescript
  protected createVariantCategoryGroup(
    elements: VariantMatrixElement[]
  ): VariantCategoryGroup {
    const parentVariantCategory = elements.length ? elements[0]?.parentVariantCategory : undefined;
```